### PR TITLE
Replace #defines with const variables and remove compiler warnings.

### PR DESCRIPTION
**Pull request by amit-cliqz**
1. Replace #defines and use const variables instead.
2. Introduce compiler flag to show warnings.
3. Fix all compiler warnings except in 3rd party libraries.

### CODE REVIEW

File: keyvi/include/keyvi/dictionary/fsa/internal/constants.h [hunk 1]
```diff
@@ -38,32 +38,30 @@ static const int KEYVI_FILE_VERSION_MIN = 2;
...
+static const size_t COMPACT_SIZE_INNER_WEIGHT_MAX_VALUE = 0xffff;
 
 // how many buckets to go left doing (brute force) search for free buckets in
 // the sparse array where the new state fits in
-#define SPARSE_ARRAY_SEARCH_OFFSET 151
+static const size_t SPARSE_ARRAY_SEARCH_OFFSET = 151;
 
 // 1 GB default memory limit for the sorter (tpie)
-#define DEFAULT_MEMORY_LIMIT_TPIE_SORT 1073741824
+static const int DEFAULT_MEMORY_LIMIT_TPIE_SORT = (1 << 30);
```
Comment (hendrikmuhs) [inline keyvi/include/keyvi/dictionary/fsa/internal/constants.h]: The type should be `size_t` 

`1 << 30` looks cryptic to me and as this isn't performance relevant I would rather keep the old style or define it by multiplies: `1 * 1024 * 1024 * 1024`

File: keyvi/tests/keyvi/dictionary/fsa/internal/memory_map_flags_test.cpp [hunk 2]
```diff
@@ -38,12 +38,13 @@ BOOST_AUTO_TEST_SUITE(MemoryMapFlagsTests)
...
 BOOST_AUTO_TEST_CASE(MemoryMapFlagsTestLazy) {
   loading_strategy_types strategy = loading_strategy_types::lazy;
-  int key_flags = MemoryMapFlags::FSAGetMemoryMapOptions(strategy);
-  int value_flags = MemoryMapFlags::ValuesGetMemoryMapOptions(strategy);
   auto key_advise_flags = MemoryMapFlags::FSAGetMemoryMapAdvices(strategy);
   auto value_advise_flags = MemoryMapFlags::ValuesGetMemoryMapAdvices(strategy);
 
 #if not defined(OS_MACOSX)
+  int key_flags = MemoryMapFlags::FSAGetMemoryMapOptions(strategy);
+  int value_flags = MemoryMapFlags::ValuesGetMemoryMapOptions(strategy);
```
Comment (hendrikmuhs) [inline keyvi/tests/keyvi/dictionary/fsa/internal/memory_map_flags_test.cpp]: good spot, never noticed as I work on linux

File: keyvi/include/keyvi/dictionary/fsa/internal/constants.h (hunk 1 above)
Comment (amit-cliqz) [reply keyvi/include/keyvi/dictionary/fsa/internal/constants.h]: Thanks for pointing out, I will fix it.

**Comment by hendrikmuhs** [comment]
Nice cleanup! Thanks @amit-cliqz 

We need of course to sort out the style problems. It seems like `memory_map_manager/flags_test.cpp` haven't been touched after the introduction of clang-format and cpplint. So you need to fix the issues there to get CI green.

May I also ask you to check the type glitch when compiling the python extension on linux, for example: 

from https://travis-ci.org/KeyviDev/keyvi/jobs/370108555:

```
In file included from /home/travis/build/KeyviDev/keyvi/python/src/cpp/keyvi/include/keyvi/dictionary/dictionary.h:37:0,
                 from _core.cpp:594:
/home/travis/build/KeyviDev/keyvi/python/src/cpp/keyvi/include/keyvi/dictionary/matching/fuzzy_matching.h: In member function ‘keyvi::dictionary::Match keyvi::dictionary::matching::FuzzyMatching::NextMatch()’:
/home/travis/build/KeyviDev/keyvi/python/src/cpp/keyvi/include/keyvi/dictionary/matching/fuzzy_matching.h:86:75: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (query_char_length_ > candidate_length() && intermediate_score > max_edit_distance_) {
                                                                           ^
/home/travis/build/KeyviDev/keyvi/python/src/cpp/keyvi/include/keyvi/dictionary/matching/fuzzy_matching.h:96:72: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
       if (traverser_ptr_->IsFinalState() && metric_ptr_->GetScore() <= max_edit_distance_) {
                                                                        ^

```

That hopefully clears all warnings together with you changes.

File: CMakeLists.txt [hunk 3]
```diff
@@ -9,6 +9,7 @@ else()
 endif()
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall")
```
Comment (narekgharibyan) [inline CMakeLists.txt]: I'd rather add `-Wall` to line 11. like
```
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -msse4.2 -Wall") 
```

File: keyvi/include/keyvi/dictionary/fsa/internal/constants.h [hunk 4]
```diff
@@ -38,32 +38,30 @@ static const int KEYVI_FILE_VERSION_MIN = 2;
...
 
 // 1 GB default memory limit for the generator
-#define DEFAULT_MEMORY_LIMIT_GENERATOR 1073741824
+static const int DEFAULT_MEMORY_LIMIT_GENERATOR = (1 << 30);
 
 // 500MB default memory limit for value stores
-#define DEFAULT_MEMORY_LIMIT_VALUE_STORE 524288000
+static const int DEFAULT_MEMORY_LIMIT_VALUE_STORE = (500 << 20);
```
[the remark this answers is not in the document]
Comment (amit-cliqz) [reply keyvi/include/keyvi/dictionary/fsa/internal/constants.h]: I will replace them with size_t. Thanks for pointing out.

**Comment by hendrikmuhs** [comment]
@amit-cliqz 

would be good if you only fix the files you have to touch, otherwise you get lost on all the changes. I see some unrelated changes, all good, but this becomes a monster PR and gets hard to review/integrate. Can you try to make smaller changes and rather create separate PR's for separate issues. E.g. a PR for 'fix formating for subdirectory xyz'. 

Your const'ifications are great, let's keep that as 1 PR, than 1 PR for -WAll, separate PR's for 'fix formating'.

**Comment by amit-cliqz** [comment]
Will create new smaller PRs, so closing this one.

**Pull request closed without merging**